src/main/java/org/prebid/cache/repository/redis/RedisPropertyConfiguration.java

```diff
@@ -1,29 +1,104 @@
 package org.prebid.cache.repository.redis;
 
+import io.lettuce.core.RedisClient;
 import io.lettuce.core.RedisURI;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.api.reactive.RedisStringReactiveCommands;
+import io.lettuce.core.cluster.RedisClusterClient;
+import io.lettuce.core.cluster.api.StatefulRedisClusterConnection;
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
 @Configuration
 @ConditionalOnProperty(prefix = "spring.redis", name = {"host"})
 @ConfigurationProperties(prefix = "spring.redis")
 public class RedisPropertyConfiguration {
+
     private String host;
-    private Integer port;
+    private long timeout;
     private String password;
+    private Boolean isCluster;
+
+    private RedisURI createRedisURI() {
+        requireNonNull(host);
+        String[] hostParams = host.split(":");
+
+        if (isNull(password)) {
+            return RedisURI.Builder.redis(hostParams[0], hostParams.length == 2 ? Integer.parseInt(hostParams[1]) : 0)
+                    .withTimeout(Duration.ofMillis(timeout))
+                    .build();
+        } else {
+            return RedisURI.Builder.redis(hostParams[0], hostParams.length == 2 ? Integer.parseInt(hostParams[1]) : 0)
+                    .withTimeout(Duration.ofMillis(timeout))
+                    .withPassword(password)
+                    .build();
+        }
+    }
+
+    private List<RedisURI> createRedisClusterURIs() {
+        requireNonNull(host);
+        return Arrays.stream(host.split(","))
+                .map(host -> {
+                    String[] hostParams = host.split(":");
+                    String hostname = requireNonNull(hostParams[0]);
+                    int port = hostParams.length == 2 ? Integer.parseInt(hostParams[1]) : 0;
+                    return isNull(password) ? RedisURI.Builder.redis(hostname, port)
```

Comment:
This expression is not easy to read and it contains duplications. Please refactor it something along the lines of:
```
final RedisURI.Builder builder = RedisURI.Builder.redis(hostname, port).withTimeout(Duration.ofMillis(timeout));
if(password != null) {
    builder..withPassword(password);
}

return builder.build();